app/domain/conjur/fetch_trusted_proxies.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Conjur
+
+  FetchTrustedProxies ||= CommandClass.new(
+    dependencies: {
+        role_cls:                 ::Role,
+        validate_account_exists:  ::Authentication::Security::ValidateAccountExists.new,
+        logger:                   Rails.logger
+    },
+    inputs: %i(account)
+  ) do
+
+    def call
+      validate_account_exists
+      fetch_trusted_proxies_list
+      delete_duplications_in_list
+      trusted_proxies_list
+    rescue => e
+      raise Errors::Conjur::TrustedProxiesMissing.new(
```

Comment:
this is something that we don't do in other places. can you elaborate on the cases where this error would occur? I am trying to see if we can avoid it and place the `raise` inside one of the methods.

Also - please at lease add a `begin` after `validate_account_exists` so we don't get a `TrustedProxiesMissing` error when the error is that the account doesn't exist.

Reply:
I think that verify_account_missing should throw another exception - it has nothing to do with valid proxies

Reply:
This is an internal action against the DB to fetch the configured trusted proxies so no matter what is the error i want that the main error in log will be on incorrect proxy configuration with an additional reason .
So validate account is also include here

Reply:
but why? if the reason of the error is that the account doesn't exist then that's the action item that the user needs to do. why bother them with a `TrustedProxiesMissing` error?

Reply:
Because from the POV of this command class the account is not related to the user, we get this account as an input and maybe in the future it will be different from some reason.
And from UX wise i do not think its bothering the customer to have more context to the `account doesn't exist` error